app/helpers/application_notifications_helper.rb

```diff
@@ -0,0 +1,16 @@
+module ApplicationNotificationsHelper
+  # Translate the event of an activity to the path of a given notification
+  #
+  # @param [Course::Notification|UserNotification] notification The notification that you want
+  #   its view path
+  # @return [String] The path of a template
+  def notification_view_path(notification)
+    if notification.is_a?(UserNotification) || notification.is_a?(Course::Notification)
```

Comment:
Why you want to check the notification type here ?  I think this can be anything who has a reference to activity